### PR TITLE
Feature/vulnanalyzer native binary

### PR DESCRIPTION
**Pull request by mehab**
Completed changes for quarkus cache implementation. Need to check on any additional conditions that we may want to test with the team

### CODE REVIEW

File: vulnerability-analyzer/src/main/java/org/acme/VulnerabilityAnalyzerTopology.java [hunk 1]
```diff
@@ -199,14 +215,34 @@ private boolean isSwidTagId(final String swidTagId) {
 
     private List<KeyValue<String, VulnerabilityResult>> analyzeOssIndex(final List<Component> components) {
         LOGGER.info("Performing OSS Index analysis for {} components: {}", components.size(), components);
-        return ossIndexAnalyzer.analyze(components).stream() // TODO: Handle exceptions
+        final var purlComponents = new MultivaluedHashMap<String, Component>();
+        for (final Component component : components) {
+            purlComponents.add(component.getPurl().getCoordinates(), component);
```
[the remark this answers is not in the document]
Comment (mehab) [reply vulnerability-analyzer/src/main/java/org/acme/VulnerabilityAnalyzerTopology.java:220]: @sonatype-lift ignore

File: vulnerability-analyzer/src/main/java/org/acme/analyzer/SnykAnalyzer.java [hunk 2]
```diff
@@ -62,15 +57,15 @@ public List<VulnerabilityResult> analyze(final List<Component> components) {
                 .flatMap(component -> analyzeComponent(component).stream())
                 .toList();
     }
-
     private List<VulnerabilityResult> analyzeComponent(final Component component) {
         final Page<Issue> issuesPage;
+            if (snykCache.as(CaffeineCache.class).getIfPresent(component.getPurl().getCoordinates()) != null) {
```
[the remark this answers is not in the document]
Comment (mehab) [reply vulnerability-analyzer/src/main/java/org/acme/analyzer/SnykAnalyzer.java:62]: @sonatype-lift ignore

File: repository-meta-analyzer/src/main/java/org/acme/RepositoryMetaAnalyzerTopology.java [hunk 3]
```diff
@@ -67,8 +71,9 @@ public RepositoryMetaAnalyzerTopology(final ComposerMetaAnalyzer composerMetaAna
                                           final NpmMetaAnalyzer npmMetaAnalyzer,
                                           final NugetMetaAnalyzer nugetMetaAnalyzer,
                                           final PypiMetaAnalyzer pypiMetaAnalyzer,
-                                          final RepoEntityRepository repoEntityRepository,
-                                          @javax.inject.Named("metaAnalyzerCache") final Cache<MetaAnalyzerCacheKey, MetaModel> cache) {
+                                          final RepoEntityRepository repoEntityRepository
+                                          //  @javax.inject.Named("metaAnalyzerCache") final Cache<MetaAnalyzerCacheKey, MetaModel> cache
```
Comment (sahibamittal) [inline repository-meta-analyzer/src/main/java/org/acme/RepositoryMetaAnalyzerTopology.java:75]: comment can be removed

File: repository-meta-analyzer/src/main/resources/application.properties [hunk 4]
```diff
@@ -52,7 +52,12 @@ quarkus.hibernate-orm.database.generation=validate
...
 
 %dev.quarkus.hibernate-orm.log.sql=true
-quarkus.hibernate-orm.physical_naming_strategy=org.hibernate.cfg.DefaultNamingStrategy
 quarkus.hibernate-orm.active=true
 
-meta.analyzer.cache.validity.period=2H
\ No newline at end of file
+quarkus.cache.enabled=true
+quarkus.cache.type=caffeine
+quarkus.cache.caffeine."snyk".expire-after-write=2H
```
Comment (sahibamittal) [inline repository-meta-analyzer/src/main/resources/application.properties:59]: need to remove these from repository-meta-analyzer -> properties

File: vulnerability-analyzer/src/main/resources/application.properties [hunk 5]
```diff
@@ -42,22 +42,34 @@ scanner.ossindex.base.url=https://ossindex.sonatype.org
...
 scanner.snyk.token=
 scanner.snyk.severity.source=NVD
-scanner.snyk.cache.validity.period=2H
-scanner.snyk.ratelimit.timeout.duration=60
-scanner.snyk.ratelimit.limit.for.period=1500
-scanner.snyk.ratelimit.limit.refresh.period=60
+
+quarkus.cache.enabled=true
+quarkus.cache.type=caffeine
+quarkus.cache.caffeine."metaAnalyzer".expire-after-write=2H
```
Comment (sahibamittal) [inline vulnerability-analyzer/src/main/resources/application.properties]: move the metaAnalyzer properties in the meta-analyzer module.

**Review by sahibamittal — APPROVED** [review]

**Pull request merged**